osu.Game/Overlays/Dashboard/CurrentlyOnlineDisplay.cs

```diff
@@ -221,6 +181,15 @@ public OnlineUserPanel(APIUser user)
                 AutoSizeAxes = Axes.Both;
             }
 
+            protected override void Update()
+            {
+                base.Update();
+
+                // TODO: we probably don't want to do this every frame.
+                var activity = metadataClient?.GetPresence(User.Id)?.Activity;
+                spectateButton.Enabled.Value = activity is UserActivity.InSoloGame or UserActivity.InMultiplayerGame or UserActivity.InPlaylistGame;
```

Comment:
I've been told in the past to not use `or` and similar constructs. Please change to use a switch.

Reply:
yeah wasn't sure about this one it's a very weird thing. sounds good.